kds/kds.go

```diff
@@ -117,17 +124,48 @@ var (
 // TCBVersion is a 64-bit bitfield of different security patch levels of AMD firmware and microcode.
 type TCBVersion uint64
 
+// TCBVersionTurin is a 64-bit bitfield of different security patch levels of AMD firmware and microcode
+// on Turin family chips.
+type TCBVersionTurin uint64
+
+// TCBVersionI is an interface for represeting the bit pattern of all TCB components' security patch
+// levels.
+type TCBVersionI interface {
+	Decompose() TCBPartsI
+	Raw() uint64
+	tcbArgs() string
+}
+
+// TCBPartsI is an interface for representing the decomposition of security patch levels of TCB components
+// of AMD chips' SEV-SNP support.
+type TCBPartsI interface {
```

Comment:
This seems like overkill to implement twice for what we know is 8 separate bytes. Certainly SetFmcSpl doesn't make sense to implement for Milan or Genoa, and the functions don't return an error type to indicate as much.

Reply:
I thought about using the "structVersion" extension to disambiguate the TCB version structures.

Would you keep the same interface model, or scrap it and go with using the structVersion?

Reply:
Given that FmcSpl is not necessarily the last SPL to be taken by a Zen* product, I wouldn't want to lock in each one of these names. Maybe we ought to be get/setting them by their KDS GET variable name. That way the different families can have different translation tables from name to SPL index and OID. That's all that's different between them.